buildSrc/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -0,0 +1,2436 @@
+/*
+ * Copyright (c) 2019 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pegasus.gradle;
+
+import com.linkedin.pegasus.gradle.PegasusOptions.IdlOptions;
+import com.linkedin.pegasus.gradle.internal.CompatibilityLogChecker;
+import com.linkedin.pegasus.gradle.tasks.ChangedFileReportTask;
+import com.linkedin.pegasus.gradle.tasks.CheckIdlTask;
+import com.linkedin.pegasus.gradle.tasks.CheckPegasusSnapshotTask;
+import com.linkedin.pegasus.gradle.tasks.CheckRestModelTask;
+import com.linkedin.pegasus.gradle.tasks.CheckSnapshotTask;
+import com.linkedin.pegasus.gradle.tasks.GenerateAvroSchemaTask;
+import com.linkedin.pegasus.gradle.tasks.GenerateDataTemplateTask;
+import com.linkedin.pegasus.gradle.tasks.GeneratePegasusSnapshotTask;
+import com.linkedin.pegasus.gradle.tasks.GenerateRestClientTask;
+import com.linkedin.pegasus.gradle.tasks.GenerateRestModelTask;
+import com.linkedin.pegasus.gradle.tasks.PublishRestModelTask;
+import com.linkedin.pegasus.gradle.tasks.TranslateSchemasTask;
+import com.linkedin.pegasus.gradle.tasks.ValidateExtensionSchemaTask;
+import com.linkedin.pegasus.gradle.tasks.ValidateSchemaAnnotationTask;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
```

Comment:
Is there anything of particular note in the fork of this plugin we should make note of?

Reply:
I'll leave some notes. Basically apply some fixes from their code comments as well as a new implementation of their incremental processing task.